python/ray/train/torch/torch_checkpoint.py

```diff
@@ -14,29 +14,72 @@
 
 @PublicAPI(stability="beta")
 class TorchCheckpoint(Checkpoint):
-    """A :py:class:`~ray.air.checkpoint.Checkpoint` with Torch-specific
-    functionality.
+    """A :class:`~ray.air.checkpoint.Checkpoint` with Torch-specific functionality.
 
-    Create this from a generic :py:class:`~ray.air.checkpoint.Checkpoint` by calling
+    Create this from a generic :class:`~ray.air.checkpoint.Checkpoint` by calling
     ``TorchCheckpoint.from_checkpoint(ckpt)``.
     """
 
+    @classmethod
+    def from_state_dict(
+        cls,
+        state_dict: Dict[str, Any],
+        *,
+        preprocessor: Optional["Preprocessor"] = None,
+    ) -> "TorchCheckpoint":
+        """Create a :class:`~ray.air.checkpoint.Checkpoint` that stores a model state
```

Comment:
First line of docstring should be 1 line please!

Reply:
I'm not sure if we should do that.

If the docstring summary is on one line, then the line is longer than 88 characters. This normally means we should shorten our summary, but the only reason the summary is long is because of the `:class:ray.air.checkpoint.Checkpoint` Sphinx directive.

So, we either break that line limit convention or the one-line docstring summary convention. Given that the goal of the one-line summary convention is to have short summaries, I think it's fine to wrap summaries when we use Sphinx directives.

What do you think?

Reply:
Also, I don't think there's any existing precedent in our code base. PyTorch wraps their summaries. For example, https://pytorch.org/docs/stable/_modules/torch/nn/modules/linear.html#Bilinear

Reply:
The formatting guidelines we use follow PEP 257:

> Multi-line docstrings consist of a summary line just like a one-line docstring, followed by a blank line, followed by a more elaborate description. The summary line may be used by automatic indexing tools; it is important that it fits on one line and is separated from the rest of the docstring by a blank line.

https://peps.python.org/pep-0257/#multi-line-docstrings


I agree though, for long sphinx directives, there doesn't seem to be an ideal solution.

Reply:
cc @richardliaw @maxpumperla for thoughts